Point_set_3/include/CGAL/Point_set_3.h

```diff
@@ -219,11 +219,9 @@ class Point_set_3
     added. If `false` (default value), the normal map can still be
     added later on (see `add_normal_map()`).
    */
-  Point_set_3 (bool with_normal_map = false) : m_base()
+  Point_set_3 () : m_base()
   {
     clear();
-    if (with_normal_map)
-      add_normal_map();
   }
```

Comment:
unintended?

Reply:
indeed, thank you

Reply:
However it seems that this parameter is never used. Would it make sense to remove it and let the user add his normal map manually?
The user already needs to add `true` manually in this constructor anyway, but it would be a breaking change

Reply:
IMO, we should store the value of the bool and if false, create the map at the first call to get the normal map. That way, we can keep the helper function static and be even more user friendly.

Reply:
You don't need a Boolean: you can use the bool conversion operator of the property map.

Reply:
You'd need it to restore the state after `clear()` (IMHO)

Reply:
Just note the current public API has the boolean parameter

Reply:
Which is perfectly compatible with what I said :)

Reply:
> You'd need it to restore the state after `clear()` (IMHO)

In Surface_mesh there were all these shenanigans about not deleting property maps if you clear a mesh, but it doesn't really matter? Either clear() deletes the property map and then the `operator bool()` returns `false` and you'll call `add_normal_map()` again; or it doesn't, `operator bool()` is `true`, and then you can simply use the normal map as before?

Reply:
it is!

Reply:
Of course it needs to be update but there is not point not to create the map on the fly if requested. I mean it's just a matter of checking a `bool` and I don't see any scenario where this will be more significant with anything done with the map.